lib/matplotlib/_mathtext.py

```diff
@@ -2031,6 +2031,12 @@ def unknown_symbol(self, s, loc, toks):
         r'overrightarrow': r'\rightarrow',
         r'overleftarrow':  r'\leftarrow',
         r'mathring':       r'\circ',
+        r'r':              r'\circ',
```

Comment:
This causes most of the errors (as the parsing confuses it with `\right` somehow...).